### PR TITLE
Add bzero, calloc, memcpy, memmove, memset, strchr, strrchr, toupper

### PR DESCRIPTION
**Pull request by hrystynaGr**

### CODE REVIEW

File: ft_calloc.c [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+/* ************************************************************************** */
+
+#include <stddef.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+void	*ft_calloc(size_t nitems, size_t size);
+
+void	*calloc(size_t nitems, size_t size)
```
Comment (anna-balan) [inline ft_calloc.c:20]: //just need change to ft_calloc

File: ft_calloc.c [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+/*                                                                            */
+/* ************************************************************************** */
+
+#include <stddef.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+void	*ft_calloc(size_t nitems, size_t size);
+
```
Comment (anna-balan) [inline ft_calloc.c:19]: needs to change name of function to ft_calloc

File: ft_calloc.c [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+void	*calloc(size_t nitems, size_t size)
+{
+	char	*mem;
+	char	*start;
+
+	mem = (char *)malloc(sizeof(size) * nitems);
+	start = mem;
+	while (nitems != 0)
```
Comment (anna-balan) [inline ft_calloc.c:27]: maybe we should add extra check for NULL?
```
if (mem == NULL)
		return (NULL);
```

File: ft_strchr.c [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+/*   By: kkubik <kkubik@student.42prague.com>       +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/01/12 22:13:18 by kkubik            #+#    #+#             */
+/*   Updated: 2023/01/12 23:06:02 by kkubik           ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include <string.h>
+#include <stdio.h>
+
```
Comment (anna-balan) [inline ft_strchr.c:15]: works as should! )

File: ft_memmove.c [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+	size_t	i;
+	char	*srcp;
+	char	*destp;
+	char	*temp;
+
+	i = 0;
+	srcp = (char *)src;
+	destp = (char *)dest;
+	//I do not know how to do this without malloc
+	temp = (char *)malloc(sizeof(char) * n);
```
Comment (anna-balan) [inline ft_memmove.c:31]: yes, we need to think about replacing malloc

File: ft_strrchr.c [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+	i = 0;
+	while (*str != '\0')
+	{
+		str++;
+		i++;
+	}
+	return (i);
+}
+
```
Comment (anna-balan) [inline ft_strrchr.c:33]: maybe we could leave strlen function, but get rid of getthere:
```
char	*ft_strrchr(const char *str, int c)
{
	size_t	len;

	len = ft_strlen(str) + 1;
	while (len >= 0)
	{
		if (str[len] == (char)c)
			return ((char *)(str + len));
		len--;
	}
	return (0);
}
```

File: ft_calloc.c (hunk 3 above)
Comment (anna-balan) [inline ft_calloc.c:29]: maybe we should add extra check for NULL?
```
if (mem == NULL)
		return (NULL);
```

**Review by anna-balan — CHANGES_REQUESTED** [review]
1. memset() with 0 arg - I think we could leave as it
_The behavior of the memset() function with a size of 0 is well defined, as long as the pointer argument is valid.

See section 7.21.1 of the C99 standard, or 7.24.1 of the C11 standard:

Where an argument declared as size_t n specifies the length of the array for a function, n can have the value zero on a call to that function.

On the other hand, the warning is a sensible one; a call like memset(s, 0, 0) is not dangerous, but it's not useful (it does nothing), and could easily indicate a programming error._
2. > Did you know another way how to use size_t type? - I think after creation of libft.h file we could include all needed libs there
3. memmove() - I'm thinking about replacing malloc(), for now haven't good solution

File: ft_memmove.c (hunk 5 above)
Comment (anna-balan) [inline ft_memmove.c]: Maybe smth like this? Please recheck me.

> Material from here - https://www.geeksforgeeks.org/memmove-in-cc/
> memmove() copies the data first to an intermediate buffer, then from the buffer to destination.

```
void  *ft_memmove(void *dest, void *src, size_t n)
{
  size_t  i;
  char  *srcp;
  char  *destp;
  char  temp;

  i = 0;
  srcp = (char *)src;
  destp = (char *)dest;
  //I do not know how to do this without malloc
  //temp = (char *)malloc(sizeof(char) * n);
  if (!dest && !src)
    return (NULL);
  while (i < n)
  {
    temp = srcp[i];
     destp[i] = temp;
    i++;
  }

  return (dest);
}
```

**Review by anna-balan — CHANGES_REQUESTED** [review]

File: ft_calloc.c (hunk 2 above)
Comment (hrystynaGr) [reply ft_calloc.c:19]: Thank you for noticing! it would be the worst mistake ever to send it like this T_T

File: ft_calloc.c (hunk 3 above)
Comment (hrystynaGr) [reply ft_calloc.c:29]: You are right! Adding it right away!

File: ft_memmove.c (hunk 5 above)
Comment (hrystynaGr) [reply ft_memmove.c]: Brilliant idea to use temp as one symbol <3

**Pull request merged**